osu.Game/Overlays/KeyBinding/KeyBindingRow.cs

```diff
@@ -197,19 +213,16 @@ public bool IsBinding
                 get { return isBinding; }
                 set
                 {
+                    if (value == isBinding) return;
                     isBinding = value;
 
-                    if (value)
+                    if (isBinding)

```

Comment:
I'd prefer if you put the special logic during `isBinding` into `updateHoverState`. Right now this logic is fragmented through various call sites (and exists in multiple places `OnHover` and `OnHoverLost`).